src/go/k8s/controllers/redpanda/cluster_controller_test.go

```diff
@@ -113,6 +113,7 @@ var _ = Describe("RedPandaCluster controller", func() {
 
 			Expect(sts.Spec.Template.Spec.Containers[0].Resources.Requests).Should(Equal(resources))
 			Expect(sts.Spec.Template.Spec.Containers[0].Resources.Limits).Should(Equal(resources))
+			Expect(sts.Spec.Template.Spec.Containers[0].Env[0]).Should(Equal(corev1.EnvVar{Name: "REDPANDA_ENVIRONMENT", Value: "kubernetes"}))
```

Comment:
maybe we should be more robust here and iterate over all ENVs looking for this one?

Reply:
Thanks for pointing this out. I will try to adjust this.

Reply:
@RafalKorepta are you going to push this fix, or want me to merge?

I think part of the confusion for me is that when people leave feedback, we typically qualify it so 

`nit:` up to the user to change
else, we expect people to change everything the reviewer commented, or find a solution for. 

before we LGTM, we should make sure all feedback of the reviewer has been `resolved` and there are no pending questions. 

otherwise, I'm confused as to what has been approved or not.

Reply:
good point, next time I won't LGTM or I'll mark it as nit.

See, we're already getting better at this :) thanks for the feedback

Reply:
Let's adjust this then

Reply:
I apologize for the oversight. I used `ContainElement` to address comment.